src/Microsoft.ML.TimeSeries/SsaAnomalyDetectionBase.cs

```diff
@@ -96,178 +114,198 @@ public abstract class SsaArguments : ArgumentsBase
             public Single DiscountFactor = 1;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "The function used to compute the error between the expected and the observed value", ShortName = "err", SortOrder = 13)]
-            public ErrorFunctionUtils.ErrorFunction ErrorFunction = ErrorFunctionUtils.ErrorFunction.SignedDifference;
+            public ErrorFunction ErrorFunction = ErrorFunction.SignedDifference;
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "The flag determing whether the model is adaptive", ShortName = "adp", SortOrder = 14)]
             public bool IsAdaptive = false;
         }
 
-        protected readonly int SeasonalWindowSize;
-        protected readonly Single DiscountFactor;
-        protected readonly bool IsAdaptive;
-        protected readonly ErrorFunctionUtils.ErrorFunction ErrorFunction;
-        protected readonly Func<Double, Double, Double> ErrorFunc;
-        protected SequenceModelerBase<Single, Single> Model;
+        internal SsaAnomalyDetectionBase Base;
 
-        public SsaAnomalyDetectionBase(SsaArguments args, string name, IHostEnvironment env)
-            : base(args.WindowSize, 0, args.Source, args.Name, name, env, args.Side, args.Martingale, args.AlertOn, args.PowerMartingaleEpsilon, args.AlertThreshold)
-        {
-            Host.CheckUserArg(2 <= args.SeasonalWindowSize, nameof(args.SeasonalWindowSize), "Must be at least 2.");
-            Host.CheckUserArg(0 <= args.DiscountFactor && args.DiscountFactor <= 1, nameof(args.DiscountFactor), "Must be in the range [0, 1].");
-            Host.CheckUserArg(Enum.IsDefined(typeof(ErrorFunctionUtils.ErrorFunction), args.ErrorFunction), nameof(args.ErrorFunction), ErrorFunctionUtils.ErrorFunctionHelpText);
-
-            SeasonalWindowSize = args.SeasonalWindowSize;
-            DiscountFactor = args.DiscountFactor;
-            ErrorFunction = args.ErrorFunction;
-            ErrorFunc = ErrorFunctionUtils.GetErrorFunction(ErrorFunction);
-            IsAdaptive = args.IsAdaptive;
-            // Creating the master SSA model
-            Model = new AdaptiveSingularSpectrumSequenceModeler(Host, args.InitialWindowSize, SeasonalWindowSize + 1, SeasonalWindowSize,
-                DiscountFactor, AdaptiveSingularSpectrumSequenceModeler.RankSelectionMethod.Exact, null, SeasonalWindowSize / 2, false, false);
-
-            StateRef = new State();
-            StateRef.InitState(WindowSize, InitialWindowSize, this, Host);
-        }
+        public SsaAnomalyDetectionBaseWrapper(SsaArguments args, string name, IHostEnvironment env) { Base = new SsaAnomalyDetectionBase(args, name, env, this); }
```

Comment:
>SsaAnomalyDetectionBaseWrapper [](start = 15, length = 30)

If this is a base class, and users don't need to instantiate it can we make the two constructors internal?